packit/cli/builds/koji_build.py

```diff
@@ -82,52 +83,62 @@ def koji(
     PATH_OR_URL argument is a local path or a URL to the upstream git repository,
     it defaults to the current working directory
     """
-    api = get_packit_api(
+    collection_api = get_packit_api(
         config=config, dist_git_path=dist_git_path, local_project=path_or_url
```

Comment:
When I started working on this I was also thinking of changing `PackitAPI` internally but then I saw this code https://github.com/packit/packit/blob/eceeeae23474d5538322e00fb6a4e0524e5e51d0/packit/cli/utils.py#L164

It seems to me that the relationship between `dg` object and `lp_upstream` object in `PackitAPI` is decided outside the `PackitAPI` class and for this reason I changed `PackitAPI` externally, creating a collection. If we can get rid of that line, I think I can do what you described above but otherwise I think it is not that easy.

Reply:
I don't think this is the case. The block of code you are referring to figures out whether Packit's working directory is an upstream Git repo or a downstream dist-git repo, and sets `lp_upstream` and `lp_downstream` accordingly (which later on are going to be used to construct `api.dg` and `api.up`.)

This should be probably updated so that when `local_project` is a dist-git repo, and there are multiple packages configured, `PackitAPI` is created so that it's limited to work only on the package corresponding to the dist-git repo.

Reply:
Ok, tomorrow I will try to change that code and I will try to create a list of  `dg` inside the `PackitAPI` object.